AdminProperties failing on duration values

AdminPropertiesTests failing w/ 

```java
    Caused by: org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Duration` not supported by default: add Module "org.apache.pulsar.shade.com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: java.util.HashMap["connectTimeout"])
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeOptionalFields(MapSerializer.java:869)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo(MapSerializer.java:760)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:720)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:35)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4568)
    	at org.apache.pulsar.shade.com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3821)
    	at org.apache.pulsar.client.impl.conf.ConfigurationDataUtils.loadData(ConfigurationDataUtils.java:66)
```

--- a/spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java
+++ b/spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java
@@ -1815,17 +1815,17 @@ public class PulsarProperties {
 		private Set<String> tlsProtocols;
 
 		/**
-		 * Connection time out.
+		 * Duration to wait for a connection to server to be established.
 		 */
-		private Duration connectTimeout = Duration.ofMinutes(1);
+		private Duration connectionTimeout = Duration.ofMinutes(1);
 
 		/**
-		 * Server response read time out.
+		 * Server response read time out for any request.
 		 */
 		private Duration readTimeout = Duration.ofMinutes(1);
 
 		/**
-		 * Server request time out.
+		 * Server request time out for any request.
 		 */
 		private Duration requestTimeout = Duration.ofMinutes(5);
 
@@ -1946,12 +1946,12 @@ public class PulsarProperties {
 			this.tlsProtocols = tlsProtocols;
 		}
 
-		public Duration getConnectTimeout() {
-			return this.connectTimeout;
+		public Duration getConnectionTimeout() {
+			return this.connectionTimeout;
 		}
 
-		public void setConnectTimeout(Duration connectTimeout) {
-			this.connectTimeout = connectTimeout;
+		public void setConnectionTimeout(Duration connectionTimeout) {
+			this.connectionTimeout = connectionTimeout;
 		}
 
 		public Duration getReadTimeout() {
@@ -2002,10 +2002,10 @@ public class PulsarProperties {
 			map.from(this::getTlsTrustStorePassword).to(properties.in("tlsTrustStorePassword"));
 			map.from(this::getTlsCiphers).to(properties.in("tlsCiphers"));
 			map.from(this::getTlsProtocols).to(properties.in("tlsProtocols"));
-			map.from(this::getConnectTimeout).to(properties.in("connectTimeout"));
-			map.from(this::getReadTimeout).to(properties.in("readTimeout"));
-			map.from(this::getRequestTimeout).to(properties.in("requestTimeout"));
-			map.from(this::getAutoCertRefreshTime).to(properties.in("autoCertRefreshTime"));
+			map.from(this::getConnectionTimeout).asInt(Duration::toMillis).to(properties.in("connectionTimeoutMs"));
+			map.from(this::getReadTimeout).asInt(Duration::toMillis).to(properties.in("readTimeoutMs"));
+			map.from(this::getRequestTimeout).asInt(Duration::toMillis).to(properties.in("requestTimeoutMs"));
+			map.from(this::getAutoCertRefreshTime).asInt(Duration::toMillis).to(properties.in("autoCertRefreshTimeMs"));
 
 			properties.putIfAbsent("serviceUrl", "http://localhost:8080");
 

--- a/spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarPropertiesTests.java
+++ b/spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarPropertiesTests.java
@@ -19,6 +19,7 @@ package org.springframework.pulsar.autoconfigure;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
 import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatRuntimeException;
 
 import java.util.Collections;
 import java.util.HashMap;
@@ -217,6 +218,10 @@ public class PulsarPropertiesTests {
 		void adminProperties() {
 			Map<String, String> props = new HashMap<>();
 			props.put("spring.pulsar.administration.service-url", "my-service-url");
+			props.put("spring.pulsar.administration.connection-timeout", "12s");
+			props.put("spring.pulsar.administration.read-timeout", "13s");
+			props.put("spring.pulsar.administration.request-timeout", "14s");
+			props.put("spring.pulsar.administration.auto-cert-refresh-time", "15s");
 			props.put("spring.pulsar.administration.tls-hostname-verification-enable", "true");
 			props.put("spring.pulsar.administration.tls-trust-certs-file-path", "my-trust-certs-file-path");
 			props.put("spring.pulsar.administration.tls-allow-insecure-connection", "true");
@@ -231,10 +236,15 @@ public class PulsarPropertiesTests {
 			bind(props);
 			Map<String, Object> adminProps = properties.buildAdminProperties();
 
-			// Verify that the props can be loaded in a ClientBuilder
-			assertThatNoException().isThrownBy(() -> PulsarAdmin.builder().loadConf(adminProps));
+			// Verify that the props can NOT be loaded directly via a ClientBuilder due to
+			// the
+			// unknown readTimeout and autoCertRefreshTime properties
+			assertThatRuntimeException().isThrownBy(() -> PulsarAdmin.builder().loadConf(adminProps)).havingCause()
+					.withMessageContaining("Unrecognized field \"autoCertRefreshTimeMs\"");
 
 			assertThat(adminProps).containsEntry("serviceUrl", "my-service-url")
+					.containsEntry("connectionTimeoutMs", 12_000).containsEntry("readTimeoutMs", 13_000)
+					.containsEntry("requestTimeoutMs", 14_000).containsEntry("autoCertRefreshTimeMs", 15_000)
 					.containsEntry("tlsHostnameVerificationEnable", true)
 					.containsEntry("tlsTrustCertsFilePath", "my-trust-certs-file-path")
 					.containsEntry("tlsAllowInsecureConnection", true).containsEntry("useKeyStoreTls", true)

--- a/spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarAdministration.java
+++ b/spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarAdministration.java
@@ -16,7 +16,6 @@
 
 package org.springframework.pulsar.core;
 
-import java.time.Duration;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.HashMap;
@@ -89,17 +88,20 @@ public class PulsarAdministration
 
 	private void loadConf(PulsarAdminBuilder builder, Map<String, Object> adminConfig) {
 		Map<String, Object> conf = new HashMap<>(adminConfig);
-		if (conf.remove("connectTimeout") instanceof Duration connectTimeout) {
-			builder.connectionTimeout((int) connectTimeout.toMillis(), TimeUnit.MILLISECONDS);
+
+		// Workaround the fact that the PulsarAdminImpl does not attempt to construct the
+		// timeout settings from the config props
+		if (conf.remove("connectionTimeoutMs") instanceof Integer connectTimeout) {
+			builder.connectionTimeout(connectTimeout, TimeUnit.MILLISECONDS);
 		}
-		if (conf.remove("readTimeout") instanceof Duration readTimeout) {
-			builder.readTimeout((int) readTimeout.toMillis(), TimeUnit.MILLISECONDS);
+		if (conf.remove("readTimeoutMs") instanceof Integer readTimeout) {
+			builder.readTimeout(readTimeout, TimeUnit.MILLISECONDS);
 		}
-		if (conf.remove("requestTimeout") instanceof Duration requestTimeout) {
-			builder.requestTimeout((int) requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
+		if (conf.remove("requestTimeoutMs") instanceof Integer requestTimeout) {
+			builder.requestTimeout(requestTimeout, TimeUnit.MILLISECONDS);
 		}
-		if (conf.remove("autoCertRefreshTime") instanceof Duration autoCertRefreshTime) {
-			builder.autoCertRefreshTime((int) autoCertRefreshTime.toMillis(), TimeUnit.MILLISECONDS);
+		if (conf.remove("autoCertRefreshTimeMs") instanceof Integer autoCertRefreshTime) {
+			builder.autoCertRefreshTime(autoCertRefreshTime, TimeUnit.MILLISECONDS);
 		}
 		builder.loadConf(conf);
 

